src/extensions/extension_manager/src/extension_manager.cpp

```diff
@@ -926,6 +938,10 @@ ADUC_Result ExtensionManager::Download(
         unsigned int timeoutInSeconds = 60 * timeoutInMinutes;
 
         result = downloadProc(entity, workflowId, workFolder.get(), timeoutInSeconds, downloadProgressCallback);
+        if (IsAducResultCodeFailure(result.ResultCode))
```

Comment:
can you make a PR that only includes this fix and then another with all of your excellent additions (UTs, etc.)